### PR TITLE
Bug/disappearing avatars

### PR DESCRIPTION
**Pull request by gfodor**
Fixes a wide variety of issues causing avatars to disappear or appear choppy.

Fixes https://github.com/mozilla/hubs/issues/1843

- Previously, we were using vertex skinning to position the hips (updated via `ik-controller`.) The net effect of this is that the displacement made to the hip (for example, the height delta needed to move the head up) was being done on GPU, so was not visible to three.js for accurate frustum culling. Three frustum culling uses the geometry's boundingSphere, and so our boundingSphere for avatars was very far off since the root hip transform was not baked into the matrixWorld of the root avatar geometry. To fix this, `ik-controller` now transforms the `AvatarRoot` entity instead of the hip element with the computed hip transform needed to properly orient the head. The hip joint transform is now fixed at 0,0,0 in local space. I kept the local references to `hips` in the naming, to make it clear that we are still doing the IK math to determine logical hip transform, but are applying it to the root avatar entity.

- Previously, we were not taking into account the camera viewfinder frustum for optimizations which disabled billboarding and IK updates on avatars that were 'not visible'. This led to visible choppiness on avatars, nametags, and other elements when taking a video recording that were in the camera viewfinder frustum but not in the player frustum (eg behind the player.) This PR updates things so that those optimizations will be disabled on entities which are in the view frustum of the currently owned camera when that camera's viewfinder is active. (Note that on mobile the viewfinder is only active when you are holding the camera, so this change will not affect overall frame performance except for when the camera is being held.)

- Previously, space bubble effects were seen in the camera viewfinder. So if you were near someone and taking a picture, they would be invisible in your camera. This updates the camera renderer so that the space bubble behavior is temporarily disabled as we render the viewfinder/recording frame.

- Fixed a bunch of logic in `camera-tools` system that was causing it to misinform callers of the currently owned camera.

Along the way, adds two new facilities to the invader system, the ability to `enable`/`disable` the invader handling explicitly, as well as the ability for an external caller to inform the invader that the element it is managing the visibility for should always be invisible.

Also, I finally tracked down why avatars stopped going translucent during invasion. I think the behavior we have now where they are hidden is better safety-wise so I didn't try to restore translucency, but the problem was that our lookup by name for the avatar mesh started failing for some reason (likely as we worked on the avatar system) and so the invader logic fell back on using visibility instead of material opacity as the invasion response.

### CODE REVIEW

**Comment by brianpeiris** [comment]
You have a typo in the issue description -- it mentions that this PR fixes #2085. Did you mean #1791?

File: src/components/camera-tool.js [hunk 1]
```diff
@@ -623,6 +623,14 @@ AFRAME.registerComponent("camera-tool", {
           }
         }
 
+        const bubbleSystem = this.el.sceneEl.systems["personal-space-bubble"];
+
+        if (bubbleSystem) {
+          for (const invader of bubbleSystem.invaders) {
+            invader.disable();
+          }
```
Comment (brianpeiris) [inline src/components/camera-tool.js]: Just wanted to note that you're using a `for...of` loop in a `tock` here. Perhaps it's not a concern if the short-lived allocations we've observed can indeed be ignored.

Comment (gfodor) [reply src/components/camera-tool.js]: good point i'll refactor

**Comment by gfodor** [comment]
Ah no bug I meant was https://github.com/mozilla/hubs/issues/1843.

I'll set up a two-VR headset test, good call. I think from a math point of view there's minimal difference here since we're just moving the local transform up in the scene graph by a level, but agree its hard to be sure without fully testing other avatars.

**Comment by gfodor** [comment]
Did some testing on quest + rift simultaneously switching between minecraft, robot, and santa avatars. Seemed to work, except santa avatar revealed this bug (which existed previously) https://github.com/mozilla/hubs/issues/2097

**Pull request merged**